src/DynamoCore/Graph/Workspaces/UndoRedo.cs

```diff
@@ -464,7 +464,7 @@ public void CreateModel(XmlElement modelData)
             {
                 NodeModel nodeModel = NodeFactory.CreateNodeFromXml(modelData, SaveContext.Undo, ElementResolver);
 
-                AddAndRegisterNode(nodeModel);
+                AddNode(nodeModel);
```

Comment:
Each `Unode()` call will also call `RequestRun();` which will Run the entire Graph.
`AddAndRegisterNode` was called for each element on the undo stack. 
`AddAndRegisterNode` also calls `RequestRun();`.
`
        internal void AddAndRegisterNode(NodeModel node, bool centered = false)
        {
            if (nodes.Contains(node))
                return;

            RegisterNode(node);

            if (centered)
            {
                var args = new ModelEventArgs(node, true);
                OnRequestNodeCentered(this, args);
            }

            AddNode(node);

            HasUnsavedChanges = true;

            RequestRun();
        }
`
Besides a call to AddNode(), it seems to do a couple other things like  `RegisterNode(node);`

Should we make the RequestRun() an optional inside `AddAndRegisterNode` and keep the call to it?

Reply:
I would keep the same functionality in `AddAndRegisterNode` except for `RequestRun`.